cast/src/lib.rs

```diff
@@ -552,6 +553,41 @@ impl SimpleCast {
         Ok(res)
     }
 
+    /// Fetches a function signature given the selector using 4byte.directory
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// async fn foo() -> eyre::Result<()> {
+    ///     let sig = Cast::fourbyte("0x32e785af").await?;
+    ///     println!("{:?}", sig);
+    ///     Ok(())
+    /// }
+    /// ```
+    pub async fn fourbyte(selector: &str) -> Result<Vec<String>> {
+        #[derive(Serialize, Deserialize, Debug)]
+        struct Decoded {
+            text_signature: String,
+        }
+
+        #[derive(Serialize, Deserialize, Debug)]
+        struct ApiResponse {
+            count: i64,
+            results: Vec<Decoded>,
+        }
+
+        let selector_clean = &selector.replace("0x", "")[..8];
```

Comment:
```suggestion
        let selector_clean = selector.strip_prefix("0x").unwrap_or(selector)[..8];
```
but this would still panic if len(selector) < 8, for which there should be a check that throws an error "invalid selector" or something

Reply:
What is the benefit of using `strip_prefix("0x").unwrap_or(selector)` instead of `replace`?